### PR TITLE
Port docker resolver fix #2364.

### PR DESCRIPTION
**Pull request by Random-Liu**
Port docker resolver fix https://github.com/containerd/containerd/pull/2364.

We should really get rid of the fork. https://github.com/containerd/cri/issues/723

/cc @abhi @dmcgowan 

Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

**Comment by Random-Liu** [comment]
We should cherrypick this, and catch on the containerd 1.1.1 release train.

File: pkg/containerd/resolver/fetcher_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2018 The Containerd Authors.
+
```
Comment (mikebrow) [inline pkg/containerd/resolver/fetcher_test.go]: Was going to say.. boiler plate mismatch.. see you already fixed it.

File: pkg/containerd/resolver/fetcher.go [hunk 2]
```diff
@@ -114,6 +116,30 @@ func (r dockerFetcher) open(ctx context.Context, u, mediatype string, offset int
...
 	}
+	if offset > 0 {
+		cr := resp.Header.Get("content-range")
+		if cr != "" {
+			if !strings.HasPrefix(cr, fmt.Sprintf("bytes %d-", offset)) {
+				return nil, errors.Errorf("unhandled content range in response: %v", cr)
+
+			}
+		} else {
+			// TODO: Should any cases where use of content range
```
Comment (mikebrow) [inline pkg/containerd/resolver/fetcher.go]: todo format..

Comment (Random-Liu) [reply pkg/containerd/resolver/fetcher.go]: This is copied from the original containerd fix. Wanna to keep change minimal so as to make it easier to maintain the fork or consolidate in the future.

Comment (mikebrow) [reply pkg/containerd/resolver/fetcher.go]: yes I saw it was a copy.. todo comment would not be much of a merge :-) but already approved skip the nit .. merge away!

**Pull request merged**